doc/source/reference/arrays.dtypes.rst

```diff
@@ -179,19 +179,14 @@ Built-in Python types
     ===================      ===============
     :class:`int`             :class:`int\_`
     :class:`bool`            :class:`bool\_`
-    :class:`float`           :class:`float\_`
-    :class:`complex`         :class:`cfloat`
+    :class:`float`           :class:`double`
+    :class:`complex`         :class:`cdouble`
```

Comment:
I think `float64` and `complex128` are better canonical names than `double` and `cdouble`, can we use those names here and elsewhere in this PR? Particularly in the docs. We're moving into a future were numpy's dtypes don't vary in bitsize by platform, let's take advantage of that and get rid of some legacy C baggage in the Python API and docs too.

I would only use `double` in contexts where it would be helpful to refer to C types.

Reply:
Sure! I renamed `double` and `cdouble` in this PR (introduced to get rid of `float_` and `cfloat` that is being removed) to these sized aliases.